sbt-plugin/src/main/scala/org/scalajs/sbtplugin/HTMLRunnerTemplate.scala

```diff
@@ -0,0 +1,97 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js sbt plugin        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+
+package org.scalajs.sbtplugin
+
+import java.io.File
+import java.io.Writer
+
+import sbt.testing.{Framework, TaskDef}
+
+import org.scalajs.core.tools.json._
+import org.scalajs.testadapter.TaskDefSerializers._
+
+/** Template for the HTML runner.
+ *
+ *  Public due to internal reasons. Do not consume.
+ */
+object HTMLRunnerTemplate {
+
+  def render(title: String, sjsFile: File, jsdepsFile: File, css: File,
+      loadedFrameworks: Map[sbt.TestFramework, Framework],
+      definedTests: Seq[sbt.TestDefinition], sysProps: Map[String, String],
+      out: Writer): Unit = {
+    // Open: <html>, <head>, <title>.
+    out.write("""
+      <!DOCTYPE html>
+      <html>
+      <head>
+        <title>""")
+    writeHtmlEscaped(title, out)
+    // Close: <title>. Write: content-type UTF8. Open: CSS.
+    out.write("""</title>
+        <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
+        <link rel="stylesheet" type="text/css" href="""")
+    writeHtmlEscaped(css.toURI.toASCIIString, out)
+    // Close: CSS. Open: <script> with __ScalaJSEnv.
+    out.write(""""/>
+        <script type="text/javascript">
+          var __ScalaJSEnv = {"javaSystemProperties":""")
+    writeJSON(sysProps.toJSON, out)
+    // Close: <script>. Open: <script> for jsdeps.
+    out.write("""};
+        </script>
+        <script type="text/javascript" src="""")
+    writeHtmlEscaped(jsdepsFile.toURI.toASCIIString, out)
+    // Close: <script>. Open: <script> for SJS.
+    out.write(""""></script>
+        <script type="text/javascript" src="""")
+    writeHtmlEscaped(sjsFile.toURI.toASCIIString, out)
+    // Close: <script>. Open: <script> for test definitions.
+    out.write(""""></script>
+        <script type="text/javascript">""")
```

Comment:
I made everything spliced. Should probably be fine. We can still optimize if it becomes an issue.